code/modules/mob/living/carbon/human/examine.dm

```diff
@@ -403,3 +403,22 @@
 				return 0
 	else
 		return 0
+
+/proc/fluid_color_type_map(var/supplied_color)
+	var/list/color_map = list(
+					COLOR_HUMAN_BLOOD = "blood",
+					COLOR_DIONA_BLOOD = "blood",
+					COLOR_SKRELL_BLOOD = "blood",
+					COLOR_VAURCA_BLOOD = "blood",
+					COLOR_IPC_BLOOD = "oil",
+					COLOR_OIL = "oil",
+					COLOR_SNOW = "snow",
+					COLOR_ASH = "ash",
+						)
+
+	var/output_text = "fluid"
+	for(var/color in color_map)
+		if(color == supplied_color)
+			output_text = color_map[supplied_color]
+			break
```

Comment:
```suggestion
	var/output_text = color_map[supplied_color] || "fluid"
```

This all can be reduced to _this_. Since the input is non-numeric and list access by key returns `null` if the key is not found.

Reply:
damn, this is pretty dope, thanks for letting me know